platform/ceph/BUILD.bazel

```diff
@@ -92,7 +92,22 @@ cmake_external(
     },
     lib_source = "@ceph//:source",
     make_commands = [
-        "make -j12",
+        # we cannot just use -j1, because that will take hours, but bazel and
+        # make both trying to run their own multithreading subsystems will
+        # cause conflicts, so we need to make sure that not too many make
+        # threads are used. and we also need to make sure that we never use
+        # less than -j1, even if there's only one processor.
+        #
+        # so we compute the number of processors available, and divide by two
+        # rounding up.
+        #
+        # if this ever accidentally becomes 'make -j' due to the expr commands
+        # failing, then it's going to be terrible and (from experience)
+        # probably hang the entire build machine for several minutes before the
+        # OOM killer gets fed up with Bazel, but I can't come up with a good
+        # way around that, and hopefully this code is correct enough that it
+        # never happens.
+        "make -j$(expr $(expr 1 + $(cat /proc/cpuinfo | cut -f 1 | grep -E '^processor$' | wc -l)) / 2)",
```

Comment:
- I don't think you need the nested `$(expr ...)` here, `expr` understands parentheses just fine.

- use `</proc/cpuinfo ...`, not `cat /proc/cpuinfo | ...`

- There are a couple ways to detect the inner command failing in bash, such as setting a variable and using an explicit test.  But at this level of complexity it might be appropriate to refactor this into a separate script.

Reply:
- Will fix
 - Will fix
 - The make_commands parameter is badly implemented, and there appears to be no way to embed a reference to an external file in it, hence inlining the logic that really deserves a separate script. I will try to update the logic to be smarter, just inline.